docs/api/cypress-api/ensure.mdx

```diff
@@ -0,0 +1,81 @@
+---
+title: Cypress.ensure
+---
+
+`Cypress.ensure.method()` is a collection of helper methods for verifying
+conditions while executing Cypress commands. They are mostly useful when writing
+[custom queries](/api/cypress-api/custom-queries) or
+[commands](/api/cypress-api/custom-commands).
+
+Most functions on `Cypress.ensure` accept `subject` argument, and assert
+something about it, throwing an error if it doesn't match. They have no return
+value; their only purpose is to ensure the state of the application and throw an
+error if it doesn't match.
+
+## Syntax
+
+```javascript
+// Command chain
+Cypress.ensure.isChildCommand(command, args, cy)​
+
+// Type of argument
+Cypress.ensure.isType(subject, type, name, cy)​
+Cypress.ensure.isElement(subject, name, cy)​
+Cypress.ensure.isWindow(subject, name, cy)
+Cypress.ensure.isDocument(subject, name, cy)​
+
+// State of DOM element
+Cypress.ensure.isAttached(subject, name, cy)​
+Cypress.ensure.isNotDisabled(subject, name)​
+Cypress.ensure.isNotHiddenByAncestors(subject, name)​
+Cypress.ensure.isNotReadonly(subject, name)​
+Cypress.ensure.isScrollable($el, name)​
+Cypress.ensure.isStrictlyVisible(subject, name)​
+Cypress.ensure.isVisible(subject, name)​
+```
```

Comment:
Why are most of the referenced as "subject" vs "$el"?

Reply:
`subject` is what they'll see in intellesense, it's how these functions are used in the Cypress repo itself (), and the function signature of a query is `(subject: Any) => Any`.

The primary use of these is to call them on a `subject`, to ensure the subject matches the conditions, so I think they should be documented that way. `$el` is extremely non-descriptive in comparison, and doesn't indicate anything unless you're familiar with our convention that '$' means "A jquery object".

Reply:
Changed the one remaining over to `subject` so they're all consistent.